lib/tbd/psi.rb

```diff
@@ -1351,16 +1546,18 @@ def process(model = nil, argh = {})
         next unless deratables.include?(id)
 
         # Evaluate current set content before processing a new linked surface.
-        is            = {}
-        is[:head    ] = set.keys.to_s.include?("head"    )
-        is[:sill    ] = set.keys.to_s.include?("sill"    )
-        is[:jamb    ] = set.keys.to_s.include?("jamb"    )
-        is[:corner  ] = set.keys.to_s.include?("corner"  )
-        is[:parapet ] = set.keys.to_s.include?("parapet" )
-        is[:party   ] = set.keys.to_s.include?("party"   )
-        is[:grade   ] = set.keys.to_s.include?("grade"   )
-        is[:balcony ] = set.keys.to_s.include?("balcony" )
-        is[:rimjoist] = set.keys.to_s.include?("rimjoist")
+        is               = {}
+        is[:head       ] = set.keys.to_s.include?("head")
+        is[:sill       ] = set.keys.to_s.include?("sill")
+        is[:jamb       ] = set.keys.to_s.include?("jamb")
+        is[:corner     ] = set.keys.to_s.include?("corner")
+        is[:parapet    ] = set.keys.to_s.include?("parapet")
+        is[:roof       ] = set.keys.to_s.include?("roof")
+        is[:party      ] = set.keys.to_s.include?("party")
+        is[:grade      ] = set.keys.to_s.include?("grade")
+        is[:balcony    ] = set.keys.to_s.include?("balcony")
+        is[:balconysill] = set.keys.to_s.include?("balconysill")
+        is[:rimjoist   ] = set.keys.to_s.include?("rimjoist")
```

Comment:
Although limited to ASHRAE 90.1 2022 (A10), the distinction between:

- _roof_ vs _parapet_ edge
- _balconysill_ vs _balcony_ vs _rimjoist_ vs _sill_

... will equally apply to all non-ASHRAE PSI factor sets (TBD built-in sets, bespoke sets on file). If sets do not make such distinctions, fall backs values are set up, e.g. _balconysill_ types fall back on _balcony_ PSI factors.